src/ExadelMentorship.BusinessLogic/Features/WeatherFeature/WeatherApiService.cs

```diff
@@ -104,5 +105,45 @@ public async Task<List<City>> GetFutureTemperatureByCoordinateAndDayQuantity(Coo
                 throw new NotFoundException($"Error: {(int)result.StatusCode}");
             }
         }
+        public async Task<MaxTempCityInfo> GetTemperatureByCityNameForMaxTemp(string name)
+        {
+            var watch = new Stopwatch();
+            watch.Start();
+            var url = $"https://api.openweathermap.org/data/2.5/weather?q={name}&appid=7e66067382ed6a093c3e4b6c22940505&units=metric";
+            var httpClient = _httpClientFactory.CreateClient();
+            HttpResponseMessage result = await httpClient.GetAsync(url);
+
+            if ((int)result.StatusCode == 404)
+            {
+                return new MaxTempCityInfo
+                {
+                    Name = name,
+                    ErrorMessage = $"City: {name} was not found"
+                };
+                //throw new NotFoundException($"City: {name} was not found");
+            }
+            else if (result.IsSuccessStatusCode)
+            {
+                var json = result.Content.ReadAsStringAsync().Result;
```

Comment:
OK... try to avoid using Task.Result at all.

Reply:
done